buildozer/targets/osx.py

```diff
@@ -47,53 +47,47 @@ def ensure_sdk(self):
         check_call(('unzip', 'master.zip'), cwd=platdir)
         check_call(('rm', 'master.zip'), cwd=platdir)
 
-    def download_kivy(self, cwd, kivy='Kivy2'):
-        self.buildozer.info('Downloading kivy...')
-        if not exists(cwd+'/'+kivy+'.7z'):
-            check_call(
-                ('curl', '-O', '-L',
-                'http://kivy.org/downloads/tests/{}.7z'.format(kivy)),
-                cwd=cwd)
-        if not exists(cwd+'/Keka.app'):
-            if exists('/Applications/Keka.app'):
-                check_call(
-                    ('cp', '-a', '/Applications/Keka.app', './Keka.app'),
-                     cwd=cwd)
-            else:
-                check_call(('curl', '-o',
-                    'http://www.kekaosx.com/release/Keka-1.0.4-intel.dmg'),
-                    cwd=cwd)
-                check_call(
-                    ('hdiutil', 'attach', 'Keka-1.0.4-intel.dmg'),
-                    cwd=cwd)
-                check_call(
-                    ('cp', '-a','/Volumes/Keka/Keka.app',
-                    './Keka.app'), cwd=cwd)
-                check_call(('hdutil', 'detach', '/Volumes/Keka'))
-        check_call(
-            ('Keka.app/Contents/Resources/keka7z',
-            'x', '{}.7z'.format(kivy)), cwd=cwd)
-        check_call(('rm', '-rf', '{}.7z'.format('Kivy2')), cwd=cwd)
-        check_call(('mv', '{}.app'.format(kivy), 'Kivy.app'),cwd=cwd)
+    def download_kivy(self, cwd, py_branch=2):
+        current_kivy_vers = '1.9.1'
+
+        if exists('/Applications/Kivy{}.app'.format(py_branch)):
+            self.buildozer.info('Kivy found in Applications dir...')
+            check_call(('cp', '-a', '/Applications/Kivy{}.app'.format(py_branch), 'Kivy.app'), cwd=cwd)
+
+        else:
+
+            if not exists(cwd+'/Kivy{}.7z'.format(py_branch)):
+                self.buildozer.info('Downloading kivy...')
+                check_call(('curl', '-L', '-o', 'Kivy{}.7z'.format(py_branch),
+                    'http://kivy.org/downloads/{}/Kivy-{}-osx-python{}.7z'.format(current_kivy_vers, current_kivy_vers, py_branch)), cwd=cwd)
+
+            if not  exists('/Applications/Keka.app'):
+                self.buildozer.info('Downloading Keka as dependency (to install Kivy)')
```

Comment:
we are downloading and installing Keka to current dir 
`check_call( ('hdiutil', 'attach', 'Keka-1.0.4-intel.dmg'), cwd=cwd)` not in /Applicaiton so it will keep checking /Applications and keep downloading kaka again.